src/onINP.ts

```diff
@@ -162,31 +245,21 @@ export const onINP = (onReport: INPReportCallback, opts?: ReportOpts) => {
 
     const handleEntries = (entries: INPMetric['entries']) => {
       entries.forEach((entry) => {
-        if (entry.interactionId) {
-          processEntry(entry);
+        // Skip `first-input` entries if any `event` entries have already
+        // been processed.
+        if (
+          entry.entryType === 'first-input' &&
+          longestInteractionList.length
+        ) {
+          return;
         }
 
-        // Entries of type `first-input` don't currently have an `interactionId`,
-        // so to consider them in INP we have to first check that an existing
-        // entry doesn't match the `duration` and `startTime`.
-        // Note that this logic assumes that `event` entries are dispatched
-        // before `first-input` entries. This is true in Chrome (the only browser
-        // that currently supports INP).
-        // TODO(philipwalton): remove once crbug.com/1325826 is fixed.
-        if (entry.entryType === 'first-input') {
```

Comment:
FYI: now that some `first-input` entries have an `interactionId` property, this logic no longer works (oops!).

The problem is since line 165 above processes all entries if they have an `interactionId`, this logic here will no longer prevent duplicate `event` and `first-input` entries from getting included in the `metric.entries` array.

Anyway, I figured out that a better solution is to just ignore `first-input` entries if *any* event timing entries have already been processed, since that means the first input was long enough to get picked up by the Event Timing API.

Reply:
See comment above, but:

> that means the first input was long enough to get picked up by the Event Timing API.

I think this is the part that misled you.  Not all event timings qualify towards FID.

(Also, technically FID criteria is not 100% INP criteria.  It's actually diverging from INP as we add features.  I'd like to re-define FID as First-Interaction, but only after FID deprecation)

Reply:
See me other comment as well... admittedly my comment here was confusing, but right now the logic will only skip `first-input` entries if an INP candidate has already been found, which means an `event` entry with an `interactionId` value has been dispatched.

Reply:
Resolving as I came up with an even more betterer way to do this...